nbgrader/apps/baseapp.py

```diff
@@ -149,35 +149,28 @@ def all_configurable_classes(self) -> TypingList[MetaHasTraits]:
             if len(app.class_traits(config=True)) > 0:
                 classes.append(app)
 
+        def _collect_configurables(module):
+            """Return a list of all configurable classes from a module"""
+
+            for name in module.__all__:
+                cls = getattr(module, name)
+                if hasattr(cls, "class_traits") and cls.class_traits(config=True):
+                    classes.append(cls)
+
         # include plugins that have configurable options
-        for pg_name in plugins.__all__:
-            pg = getattr(plugins, pg_name)
-            if pg.class_traits(config=True):
-                classes.append(pg)
+        _collect_configurables(plugins)
 
         # include all preprocessors that have configurable options
-        for pp_name in preprocessors.__all__:
-            pp = getattr(preprocessors, pp_name)
-            if len(pp.class_traits(config=True)) > 0:
-                classes.append(pp)
+        _collect_configurables(preprocessors)
 
-        # include all the exchange actions
-        for ex_name in exchange.__all__:
-            ex = getattr(exchange, ex_name)
-            if hasattr(ex, "class_traits") and ex.class_traits(config=True):
-                classes.append(ex)
+        # include all the abstract exchange actions
+        _collect_configurables(exchange)
 
         # include all the default exchange actions
-        for ex_name in exchange.default.__all__:
-            ex = getattr(exchange, ex_name)
-            if hasattr(ex, "class_traits") and ex.class_traits(config=True):
-                classes.append(ex)
+        _collect_configurables(exchange.default)
```

Comment:
```suggestion
        _collect_configurables(exchange.default.exchange)
```

Since exchange is the only *default* class with dedicated traits, maybe it can be specified when calling the `_collect_configurables` function, to avoid duplicate all the other class traits.
But maybe I'm wrong, I don't know a lot about traitlets.

Reply:
Unfortunately, we can't include `default` without including `abc` as traitlets traverses up the `__bases__` and I'm not sure how to work around that. 

I think the way to make things more clear here would be to rename the actual classes in `abc`.

Reply:
After digging into this more, I think the duplication is just the nature of the beast. I have a new proposal to make the distinction more clear, which is to actually rename the `abc` classes.

When they're used in the nbgrader codebase, we already do `from exchange.abc import Exchange as ABCExchange`, and if you search GitHub, so does everyone else: https://github.com/search?q=from+nbgrader.exchange+import+&type=code

By renaming the actual class, we can get rid of these import aliases without touching the implementation, _and_ create distinction in the generated config file.